src/librustc_lint/types.rs

```diff
@@ -593,6 +618,12 @@ impl<'a, 'tcx> ImproperCTypesVisitor<'a, 'tcx> {
 
         match self.check_type_for_ffi(&mut FxHashSet(), ty) {
             FfiResult::FfiSafe => {}
+            FfiResult::FfiPhantom => {
+                self.cx.span_lint(IMPROPER_CTYPES,
+                                  sp,
+                                  &format!("found zero-sized type composed only \
+                                            of phantom-data in a foreign-function."));
```

Comment:
@emilio why is it important to highlight the fact that the zero-sized type consists only of phantom data? Isn't it true that *any* zero-sized type is suspicious, and that is the real problem here?